test/old/testdir/test_signs.vim

```diff
@@ -1509,42 +1509,46 @@ func Test_sign_priority()
               \ {'id' : 1, 'name' : 'sign1', 'lnum' : 4, 'group' : '',
               \ 'priority' : 20}],
               \ s[0].signs)
+
+  " Nvim: signs are always sorted lnum->priority->sign_id->last_modified
```

Comment:
Hmm, if this order is followed, then `test_termdebug.vim` should need the following change:
```diff
diff --git a/test/old/testdir/test_termdebug.vim b/test/old/testdir/test_termdebug.vim
index 98a4bd321..7fa162c3c 100644
--- a/test/old/testdir/test_termdebug.vim
+++ b/test/old/testdir/test_termdebug.vim
@@ -69,8 +69,7 @@ func Test_termdebug_basic()
         \  'group': 'TermDebug'},
         \ {'lnum': 9, 'id': 1014, 'name': 'debugBreakpoint1.0',
         \  'priority': 110, 'group': 'TermDebug'}],
-        "\ sign_getplaced('', #{group: 'TermDebug'})[0].signs)})
-        \ sign_getplaced('', #{group: 'TermDebug'})[0].signs->reverse())})
+        \ sign_getplaced('', #{group: 'TermDebug'})[0].signs)})
   Finish
   call Nterm_wait(gdb_buf)
   redraw!
```

But currently it doesn't.

Reply:
Not familiar with this test and I have no idea how it bypasses the qsort order. Why was it reversed in the first place? I can't reproduce it either with:
```vim
call setline(1, ['a', 'b', 'c']->repeat(3))
call sign_define([{'name':'debugPC', 'text':'>>'}, {'name':'debugBreakpoint1.0', 'text':'xx'}])
call sign_place(12, 'TermDebug', 'debugPC', bufnr(''), {'lnum':9, 'priority':110})
call sign_place(1014, 'TermDebug', 'debugBreakpoint1.0', bufnr(''), {'lnum':9, 'priority':110})
echo sign_getplaced(bufnr(''), {'group':'*'})
```

Reply:
Even if I run

    gdb --args <build/bin/nvim> -u unix.vim -U NONE -i NONE --noplugin --headless -u NONE --cmd 'set shortmess-=F' -S runtest.vim test_termdebug.vim

and break in `sign_list_append_info()` I see the expected sign order so I have no idea how this test passes with the incorrect order currently.

Reply:
> Hmm, if this order is followed, then test_termdebug.vim should need the following change:

Wait no isn't this just false? The expected results lists them in reverse order so reversing `sign_getplaced()` will make them equal...

Reply:
Ah, it's descending order? Hard to notice.

Reply:
Yes, ascending line number, descending priority and ID. That doesn't change in this PR. Only thing that changes is that modifying an already placed sign no longer gives it a higher priority in the sign list (see https://github.com/neovim/neovim/pull/25724#discussion_r1368366026).